packages/typespec-test/test/openai_modular/generated/typespec-ts/src/utils/serializeUtil.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  ChatRequestUserMessage as ChatRequestUserMessageRest,
+  ChatRequestAssistantMessage as ChatRequestAssistantMessageRest,
+  ChatRequestToolMessage as ChatRequestToolMessageRest,
+  ChatRequestMessage as ChatRequestMessageRest,
+  ChatMessageImageContentItem as ChatMessageImageContentItemRest,
+  ChatMessageContentItem as ChatMessageContentItemRest,
+} from "../rest/index.js";
+import {
+  ChatRequestUserMessage,
+  ChatRequestAssistantMessage,
+  ChatRequestToolMessage,
+  ChatRequestMessage,
+  ChatMessageImageContentItem,
+  ChatMessageContentItem,
+} from "../models/models.js";
+
+/** serialize function for ChatRequestUserMessage */
+function serializeChatRequestUserMessage(
+  obj: ChatRequestUserMessage
+): ChatRequestUserMessageRest {
+  return {
+    role: obj["role"],
+    content: obj["content"] as any,
+    name: obj["name"],
+  };
+}
+
+/** serialize function for ChatRequestAssistantMessage */
+function serializeChatRequestAssistantMessage(
+  obj: ChatRequestAssistantMessage
+): ChatRequestAssistantMessageRest {
+  return {
+    role: obj["role"],
+    content: obj["content"],
+    name: obj["name"],
+    tool_calls: obj["toolCalls"],
+    function_call: !obj.functionCall
+      ? undefined
+      : {
+          name: obj.functionCall?.["name"],
+          arguments: obj.functionCall?.["arguments"],
+        },
+  };
+}
+
+/** serialize function for ChatRequestToolMessage */
+function serializeChatRequestToolMessage(
+  obj: ChatRequestToolMessage
+): ChatRequestToolMessageRest {
+  return {
+    role: obj["role"],
+    content: obj["content"],
+    tool_call_id: obj["toolCallId"],
+  };
+}
+
+/** serialize function for ChatRequestMessage */
+export function serializeChatRequestMessageUnion(
+  obj: ChatRequestMessage
+): ChatRequestMessageRest {
+  switch (obj.role) {
+    case "user":
+      return serializeChatRequestUserMessage(obj as ChatRequestUserMessage);
```

Comment:
because our current implementation adds the polymorphic base into the union, we can't tell if it's `ChatRequestUserMessage` or `ChatRequestMessageParent`  type if the discriminator value is `user`, we have to add a type casting here.